# Patch to fix unintialized offset variable
#
# Copyright (C) 2006 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license version 2.1
#
--- gtkhtml-3.9.1.orig/src/htmlengine.c	2006-08-22 14:35:14.784096911 +0300
+++ gtkhtml-3.9.1/src/htmlengine.c	2006-08-22 14:36:28.342772550 +0300
@@ -6217,9 +6217,9 @@
 html_engine_focus (HTMLEngine *e, GtkDirectionType dir)
 {
 	if (e->clue && (dir == GTK_DIR_TAB_FORWARD || dir == GTK_DIR_TAB_BACKWARD)) {
-		HTMLObject *cur;
-		HTMLObject *focus_object;
-		gint offset;
+		HTMLObject *cur = NULL;
+		HTMLObject *focus_object = NULL;
+		gint offset = 0;
 
 		focus_object = html_engine_get_focus_object (e, &offset);
 		if (focus_object && html_object_is_embedded (focus_object)
@@ -6228,11 +6228,29 @@
 			return TRUE;
 
 		if (focus_object)
+		{
 			cur = move_focus_object (focus_object, &offset, e, dir);
+		}
 		else
+		{
 			cur = dir == GTK_DIR_TAB_FORWARD
 				? html_object_get_head_leaf (e->clue)
 				: html_object_get_tail_leaf (e->clue);
+			if (HTML_IS_TEXT (cur)) {
+				if (dir == GTK_DIR_TAB_FORWARD)
+					html_text_first_link_offset (
+						HTML_TEXT (cur), &offset);
+				else
+					html_text_last_link_offset (
+						HTML_TEXT (cur), &offset);
+			}
+			else
+			{
+				offset = dir == GTK_DIR_TAB_FORWARD
+					? 0
+					: html_object_get_length (cur);
+			}
+		}
 
 		while (cur) {
 			/* printf ("try child %p\n", cur); */
